py4DSTEM/process/diffraction/crystal_calibrate.py

```diff
@@ -0,0 +1,465 @@
+                                            (approximately) calibrated, this is the scaling factor to
+                                            correct that existing calibration.
+        bragg_k_power (float):              Input Bragg peak intensities are multiplied by k**bragg_k_power
+                                            to change the weighting of longer scattering vectors
+        bragg_intensity_power (float):      Input Bragg peak intensities are raised power **bragg_intensity_power.
+        k_min (float):                      min k value for fitting range (Å^-1)
+        k_max (float):                      max k value for fitting range (Å^-1)
+        k_step (float):                     step size of k in fitting range (Å^-1)
+        k_broadening (float):               Initial guess for Gaussian broadening of simulated pattern (Å^-1)
+        fit_all_intensities (bool):         Set to true to allow all peak intensities to change independently
+                                            False forces a single intensity scaling.
+        verbose (bool):                     Output the calibrated pixel size.
+        plot_result (bool):                 Plot the resulting fit.
+        figsize (list, tuple, np.ndarray)   Figure size of the plot.
+        returnfig (bool):                   Return handles figure and axis
+
+    Returns:
+        bragg_peaks_cali (BraggVectors):    Bragg vectors after calibration
+        fig, ax (handles):                  Optional figure and axis handles, if returnfig=True.
+
+    """
+
+    assert hasattr(self, "struct_factors"), "Compute structure factors first..."
+    
+    #Prepare experimental data
+    k, int_exp = self.calculate_bragg_peak_histogram(
+        bragg_peaks,
+        bragg_k_power,
+        bragg_intensity_power,
+        k_min,
+        k_max,
+        k_step
+    )
+
+    # Perform fitting
+    def fit_profile(k, *coefs):
+        scale_pixel_size = coefs[0]
+        k_broadening = coefs[1]
+        int_scale = coefs[2:]
+
+        int_sf = calc_1D_profile(
+            k,
+            self.g_vec_leng * scale_pixel_size,
+            self.struct_factors_int,
+            k_broadening=k_broadening,
+            int_scale=int_scale,
+            normalize_intensity=False,
+        )
+        return int_sf
+
+    if fit_all_intensities:
+        coefs = (
+            scale_pixel_size,
+            k_broadening,
+            *tuple(np.ones(self.g_vec_leng.shape[0])),
+        )
+        bounds = (0.0, np.inf)
+        popt, pcov = curve_fit(fit_profile, k, int_exp, p0=coefs, bounds=bounds)
+    else:
+        coefs = (scale_pixel_size, k_broadening, 1.0)
+        bounds = (0.0, np.inf)
+        popt, pcov = curve_fit(fit_profile, k, int_exp, p0=coefs, bounds=bounds)
+
+    scale_pixel_size = popt[0]
+    k_broadening = popt[1]
+    int_scale = np.array(popt[2:])
+
+    # Make a copy of bragg_peaks, apply correction
+    bragg_peaks_cali = deepcopy(bragg_peaks)
```

Comment:
please change to `bragg_peaks_cali = bragg_peaks.copy()`.  This previously had a bug, which i imagine is why deepcopy is being used, but the bug should now be patched, and it would be good to avoid deepcopy on these complex / linked datastructures if possible